### DISCUSSION TITLE
Local SB Instance Unable to Raise Logs

### DISCUSSION THREAD

**Discussion by sidouglas** [supabase/supabase#27451] [Questions] [answered]
As a super_admin, I want to be able to view the contents of the public users table. Only users with users.all should have permission.

Problem: no rows are returned even though user is a super_admin

I am a supabase noob, so I followed these instructions:

Created a user table as per here:
https://supabase.com/docs/guides/auth/managing-user-data

Followed directions here: https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac

Added change to toml file.
```
[auth.hook.custom_access_token]
enabled = true
uri = "pg-functions://postgres/public/custom_access_token_hook"

[analytics]
enabled = true
port = 54327
vector_port = 54328
backend = "postgres"
```

role_permissions - note super_admin has users.all
https://shottr.cc/s/Watc/SCR-20240620-u97.png

user_roles 
https://shottr.cc/s/W2zt/SCR-20240620-uaxp.png

users
https://shottr.cc/s/WnUX/SCR-20240620-ud0p.png

jwt  - with the correct role applied.
https://shottr.cc/s/WYqx/SCR-20240622-h4rp.png

I have updated the authorize function because nothing is being returned. No errors, no messages.

```
create or replace function public.authorize(
  requested_permission app_permission
)
returns boolean as $$
declare
  bind_permissions int;
  user_role public.app_role;
begin
  -- Fetch user role once and store it to reduce number of calls
  select (auth.jwt() ->> 'user_role')::public.app_role into user_role;
  RAISE LOG 'User role: %', user_role;

  select count(*)
  into bind_permissions
  from public.role_permissions
  where role_permissions.permission = requested_permission
    and role_permissions.role = user_role;
  RAISE LOG 'Bind permissions count: %', bind_permissions;

  RAISE LOG 'Requested permission: %, Authorized: %', requested_permission, (bind_permissions > 0);

  return bind_permissions > 0;
end;
$$ language plpgsql stable security definer set search_path = '';
```
1. In the browser i go the my nextjs application and refresh that page which triggers a `select * from users` query. This must in turn call through to postgres and therefore the authorize fn.
2. The data returned is an empty array.
3. I check the logs in my local supabase studio and see no logged messages.

https://shottr.cc/s/WZ8W/SCR-20240622-gvrp.png

Expect to see something logged.

**Comment by sidouglas** [comment]
I found this article: https://supabase.com/docs/guides/platform/logs?queryGroups=source&source=postgres_logs#configuring-pgauditlog and I get warnings raised.

```
172.26.0.9 2024-06-23 09:12:25.528 UTC [64596] authenticator@postgres LOG:  User role: <NULL>
172.26.0.9 2024-06-23 09:12:25.529 UTC [64596] authenticator@postgres LOG:  Requested permission: users.all, Authorized: f
```
Now completely stumped. My JWT is correct as per https://shottr.cc/s/WYqx/SCR-20240622-h4rp.png, there's a `user_role` of `super_admin` but according the authorize fn that's [copy and pasted out of the docs](https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac?queryGroups=language&language=plpgsql#accessing-custom-claims-in-rls-policies), its null. 🤷

My types: https://shottr.cc/s/WXWU/SCR-20240623-saup.png
I know that [you can't impersonate a user while using this jwt custom claims](https://github.com/orgs/supabase/discussions/22482), can anyone in the community assist?

**Reply by GaryAustin1** [reply]
If you log auth.jwt()->>'user_role' what do you get?  Also log auth.uid() to see if you really have a user jwt at all?

**Comment by sidouglas** [comment, marked as answer]
After a pile of digging, its a result of how we were calling `createServerClient()` on the server.  Nothing to do with supabase 🤣. I updated to it to be a singleton and now it works as expected. 

I guess this can be marked resolved because of PEBKAC and I managed to raise logs within Supabase.
